app/services/coronavirus_pages/timeline_entry_builder.rb

```diff
@@ -0,0 +1,29 @@
+class CoronavirusPages::TimelineEntryBuilder
+  def create_timeline_entries
+    if timeline_entries_from_yaml.any?
+      TimelineEntry.transaction do
+        coronavirus_page.timeline_entries.delete_all
+
+        timeline_entries_from_yaml.each do |entry|
+          TimelineEntry.create!(
+            heading: entry["heading"],
+            content: entry["paragraph"],
+            position: entry["position"],
+            coronavirus_page_id: coronavirus_page.id,
+          )
+        end
+      end
+    end
+  end
+
+private
+
+  def timeline_entries_from_yaml
+    @github_data ||= YamlFetcher.new(coronavirus_page.raw_content_url).body_as_hash
+    @github_data["content"]["timeline"]["list"]
```

Comment:
Minor - this looks like a dig: https://ruby-doc.org/core-2.3.0_preview1/Hash.html#method-i-dig

Reply:
yep, excellent shout